### PR TITLE
displayio api updates for 9.x+

### PR DESCRIPTION
**Pull request by FoamyGuy**
This supersedes https://github.com/adafruit/Adafruit_Learning_System_Guides/pull/2429

I've begun the process of updating code to use the new displayio API. 

As I'm working through these I'm also checking in the Learn Guides and making a seperate list of ones that have non-github embedded code in the guide that will need to be changed.

### CODE REVIEW

**Comment by tannewt** [comment]
Please ping me when you are ready for a review. Thanks for doing this!

**Comment by FoamyGuy** [comment]
@tannewt I think this is ready now. This covers all of the instances that I found with git grep. 

I also have a list of guides (and some page URLs) for ones that include code in non-embedded blocks so will require updating within Learn directly.

```
4x4_MIDI_Messenger
CLUE_BBQ
CLUE_Dice_Roller
CLUE_Egg_Drop
CLUE_Hand_Wash_Timer
CLUE_Servo_Barometer
CircuitPython_Display_Text
CircuitPython_OLED_Watch
CircuitPython Sharp Displayio / https://learn.adafruit.com/adafruit-sharp-memory-display-breakout/circuitpython-displayio-usage
CircuitPython_Slider / https://learn.adafruit.com/circuitpython-motorized-camera-slider/code-walkthrough
CircuitPython Displayio / https://learn.adafruit.com/circuitpython-display-support-using-displayio/tilegrid-and-group
CircuitPython Sailor Moon / https://learn.adafruit.com/sailor-moon-star-locket/circuitpython-code-walkthrough
Clue Scale / https://learn.adafruit.com/clue-coffee-scale/code-details
Clue shot timer / https://learn.adafruit.com/clue-shot-timer/code-the-shot-timer
Clue step counter / https://learn.adafruit.com/clue-step-counter-st-lsm6ds33/circuitpython-code-walkthrough
Desktop Air Monitor / https://learn.adafruit.com/aqi-case/coding-the-aqi-case
Eink Quickstart / https://learn.adafruit.com/quickstart-using-adafruit-eink-epaper-displays-with-circuitpython/example-a-name-badge
ESP32-S2 Mini Weather Station / https://learn.adafruit.com/mini-weather-station-esp32-s2-tft/pyleap-mini-weather-station-code
ESP32-S2 Reverse TFT Digital Clocl / https://learn.adafruit.com/esp32-s2-tft-digital-clock-display-featuring-blanka-chan/digital-clock-display-code
ESP32-S2 Azure / https://learn.adafruit.com/getting-started-with-microsoft-azure-and-circuitpython/coding-the-feather-esp32-s2-tft-azure-display
FunHouse Home Assist / https://learn.adafruit.com/using-the-adafruit-funhouse-with-home-assistant/code-the-sensor
Funhouse IOT Hub / https://learn.adafruit.com/adafruit-io-hub-with-the-adafruit-funhouse/code-run-through
Funhouse LIFX / https://learn.adafruit.com/funhouse-motion-detecting-lighting-for-lifx-bulbs/code-walkthrough
Funhouse Mailslot / https://learn.adafruit.com/funhouse-mail-slot-detector/code-the-funhouse-mail-detector
Funhouse Motion Outlet / https://learn.adafruit.com/motion-activated-outlet-with-the-adafruit-funhouse/coding-the-motion-sensor
Funhouse pet bowl / https://learn.adafruit.com/pet-bowl-water-level-sensing-with-the-funhouse-and-home-assistant/coding-the-water-sensor
Hallowing cat toy / https://learn.adafruit.com/hallowing-cat-toy/code
IOT Party Parrot / https://learn.adafruit.com/iot-twitter-listener-party-parrot/animating-sprites-with-circuitpython
Kitty Paw Pad / https://learn.adafruit.com/kitty-toe-bean-paw-keypad-color-tct/circuitpython-code-walkthrough
LCARS / https://learn.adafruit.com/star-trek-lcars-display/code-the-lcars-display
LED Matrix scoreboard / https://learn.adafruit.com/led-matrix-scoreboard/led-matrix-scoreboard-code-walkthrough
midi melody maker / https://learn.adafruit.com/midi-melody-maker/circuitpython-code-walkthrough
MAcropad rpc / https://learn.adafruit.com/macropad-remote-procedure-calls-over-usb-to-control-home-assistant/macropad-code
magtag christmas / https://learn.adafruit.com/magtag-daily-christmas-countdown/circuitpython-code-walkthrough
magtag project selector / https://learn.adafruit.com/adafruit-magtag-project-selector/code-run-through
matrix portal handbag / https://learn.adafruit.com/matrix-portal-rgb-display-handbag/code-the-slideshow
matrix portal s3 espn api / https://learn.adafruit.com/led-matrix-sports-scoreboard/code-the-scoreboard
matrix portal tip jar / https://learn.adafruit.com/matrix-portal-money-sensing-tip-jar/code-with-circuitpython
matrix animation player / https://learn.adafruit.com/pixel-art-matrix-display/code-the-sprite-sheet-animation-display
minesweeper / https://learn.adafruit.com/circuitpython-pyportal-minesweeper-game/the-board
ov5640 breakout / https://learn.adafruit.com/adafruit-ov5640-camera-breakout/lcd-mirror-demo-2
ocean resin light box / https://learn.adafruit.com/ocean-epoxy-resin-lightbox-with-rgb-led-matrix-image-scroller/code-explanation
octoprint mqtt / https://learn.adafruit.com/circuitpython-octoprint-controller-and-monitor/coding-the-octoprint-status-viewer
pep talk / https://learn.adafruit.com/clue-and-magtag-pep-talk-generator/code-the-clue-pep-talk-generator
PicoW httpserver / https://learn.adafruit.com/pico-w-http-server-with-circuitpython/code-the-pico-w-http-server
Pico Midi_Fighter / https://learn.adafruit.com/raspberry-pi-pico-led-arcade-button-midi-controller-fighter/circuitpython-code-walkthrough
improved thermal cam / https://learn.adafruit.com/improved-amg8833-pygamer-thermal-camera/import-and-initialize
Thermal cam / https://learn.adafruit.com/pygamer-thermal-camera-amg8833/import-and-initialize
pyportal calculator / https://learn.adafruit.com/pyportal-calculator-using-the-displayio-ui-elements/circuitpython-code
pyportal floppy / https://learn.adafruit.com/a-floppy-thumb-drive-with-a-color-file-icon-display/circuitpython-code
pyportal LIFX controller / https://learn.adafruit.com/pyportal-lifx-lighting-controller/code-walkthrough
pyportal mqtt control / https://learn.adafruit.com/pyportal-mqtt-sensor-node-control-pad-home-assistant/code-breakdown
pyportal phillips hue / https://learn.adafruit.com/pyportal-philips-hue-lighting-controller/code-walkthrough
pyportal remote / https://learn.adafruit.com/pyportal-roku-remote/code-run-through
pyportal user interface / https://learn.adafruit.com/making-a-pyportal-user-interface-displayio/display
pyportal winamp / https://learn.adafruit.com/pyportal-winamp-mp3-player/code-walk-through
rgbmatrix slot machine / https://learn.adafruit.com/rgb-matrix-slot-machine/circuitpython-code-walkthrough
tombstone matrix prop / https://learn.adafruit.com/tombstone-matrix-portal/code-the-sprite-sheet-animation-display
vertical garden barometer / https://learn.adafruit.com/clue-vertical-garden-weather-visualizer/customizing-your-code
walkman mp3 / https://learn.adafruit.com/walkmp3rson-personal-mp3-tape-player/code-the-walkmp3rson
literary clock magtag / https://learn.adafruit.com/magtag-literary-quote-clock/code-the-magtag-quote-clock

```

File: CLUE_Metal_Detector/code.py [hunk 1]
```diff
@@ -662,10 +662,10 @@ def sample_sum(pin, num):
         if not screen_on or opt == 2:  # Screen toggle
             screen_on = not screen_on
             if screen_on:
-                display.show(screen_group)
+                display.root_group = screen_group
                 display.brightness = 1.0
             else:
-                display.show(None)
```
Comment (tannewt) [inline CLUE_Metal_Detector/code.py:668]: Do we want to show the terminal here? We'll need to pass that instead if so.

File: Clue_Shot_Timer/code.py [hunk 2]
```diff
@@ -150,7 +150,7 @@ def picker(current):
                     return current
         display.refresh()
 
-    display.show(None)
+    display.root_group = None
```
Comment (tannewt) [inline Clue_Shot_Timer/code.py:153]: Here too.

File: FunHouse_HA_Companion/code.py [hunk 3]
```diff
@@ -22,7 +22,7 @@
 
 # Don't display the splash yet to avoid
 # redrawing labels after each one is added
-funhouse.display.show(None)
+funhouse.display.root_group = None
```
Comment (tannewt) [inline FunHouse_HA_Companion/code.py:25]: Here too.

File: FunHouse_IOT_Hub/iot_hub/code.py [hunk 4]
```diff
@@ -101,11 +101,11 @@
 # pylint: disable=unused-argument
 def rgb(last):
     """ Function for when the rgb screen is active """
-    display.show(None)
+    display.root_group = None
```
Comment (tannewt) [inline FunHouse_IOT_Hub/iot_hub/code.py:104]: Here and below too.

File: FunHouse_LIFX/code.py [hunk 5]
```diff
@@ -59,7 +59,7 @@ def set_label_color(conditional, index, on_color):
 
 
 # Create the labels
-funhouse.display.show(None)
+funhouse.display.root_group = None
```
Comment (tannewt) [inline FunHouse_LIFX/code.py:62]: Here too.

File: FunHouse_Mailslot/code.py [hunk 6]
```diff
@@ -23,13 +23,13 @@
 funhouse.peripherals.dotstars.fill(AMBER)
 
 # Create the labels
-funhouse.display.show(None)
+funhouse.display.root_group = None
```
Comment (tannewt) [inline FunHouse_Mailslot/code.py:26]: Here too.

File: FunHouse_Motion_Outlet/code.py [hunk 7]
```diff
@@ -68,7 +68,7 @@ def timeleft():
 funhouse.peripherals.dotstars.fill(0)
 outlet = digitalio.DigitalInOut(board.A0)
 outlet.direction = digitalio.Direction.OUTPUT
-funhouse.display.show(None)
+funhouse.display.root_group = None
```
Comment (tannewt) [inline FunHouse_Motion_Outlet/code.py:71]: Here too.

File: FunHouse_Pet_Bowl_Sensor/code.py [hunk 8]
```diff
@@ -62,7 +62,7 @@ def bowl_level_display(level):
 water_enable = digitalio.DigitalInOut(board.A0)
 water_enable.switch_to_output()
 water_level_sensor = analogio.AnalogIn(board.A1)
-funhouse.display.show(None)
+funhouse.display.root_group = None
```
Comment (tannewt) [inline FunHouse_Pet_Bowl_Sensor/code.py:65]: Here also

**Review by tannewt — CHANGES_REQUESTED** [review]
One question about showing None. Looks good otherwise. All comments just mark the spots I saw.

**Comment by FoamyGuy** [comment]
@tannewt Thank you! I forgot about the different behavior from that. 

Latest commit changes all of those instances to use `CIRCUITPYTHON_TERMINAL`

**Review by tannewt — APPROVED** [review]
Thanks!

**Comment by FoamyGuy** [comment]
I've just wrapped up going through the list of Learn Guides in the prior comment and updated instances of `show()` to `root_group =` in non-embedded code and other text. 

I believe everything is updated everywhere now for the new version

**Pull request merged**